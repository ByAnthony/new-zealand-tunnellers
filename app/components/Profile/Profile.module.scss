@use "app/variables.module.scss";

.container {
  @include variables.component-margin;

  margin-top: variables.$l-margin;

  @include variables.tablet {
    margin-top: variables.$xl-margin;
  }

  @include variables.desktop {
    margin-top: variables.$xxl-margin;
  }
}

.profile {
  @include variables.component-margin;

  @include variables.desktop {
    display: flex;
    margin: 0 auto;
    max-width: variables.$inner-width;
    gap: variables.$l-margin;
  }

  .flex-summary {
    @include variables.desktop {
      width: 40%;
    }
  }

  .summary {
    @include variables.desktop {
      flex-grow: 1;
      text-align: left;
    }
  }

  .flex-diary {
    @include variables.desktop {
      width: 60%;
    }
  }
}
