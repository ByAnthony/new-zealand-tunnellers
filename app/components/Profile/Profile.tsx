"use client";

import { displayBiographyDates } from "../../utils/helpers/roll";
import { today } from "../../utils/helpers/date";

import { HowToCite } from "../HowToCite/HowToCite";
import { ProfileDiary } from "./ProfileDiary/ProfileDiary";
import { ProfileImageSource } from "./ProfileImageSource/ProfileImageSource";
import { ProfileSources } from "./ProfileSources/ProfileSources";
import { ProfileSummary } from "./ProfileSummary/ProfileSummary";
import { Title } from "../Title/Title";
import { TunnellerProfile } from "../../types/tunneller";

import STYLES from "./Profile.module.scss";

type Props = {
  tunneller: TunnellerProfile;
};

export function Profile({ tunneller }: Props) {
  return (
    <>
      <div className={STYLES.container}>
        <div className={STYLES.header}>
          <div className={STYLES.link}>
            <a href="/tunnellers">Tunnellers</a>
          </div>
          <Title
            name={tunneller.summary.name}
            subTitle={displayBiographyDates(
              tunneller.summary.birth,
              tunneller.summary.death,
            )}
          />
        </div>
      </div>
      <div className={STYLES.profile}>
        <div className={STYLES["flex-summary"]}>
          <div className={STYLES.summary}>
            <ProfileSummary
              summary={tunneller.summary}
              embarkationUnit={tunneller.militaryYears.embarkationUnit}
              enlistment={tunneller.militaryYears.enlistment}
              image={tunneller.image}
            />
          </div>
        </div>
        <div className={STYLES["flex-diary"]}>
          <ProfileDiary
            tunnellerId={tunneller.id}
            origins={tunneller.origins}
            preWarYears={tunneller.preWarYears}
            militaryYears={tunneller.militaryYears}
            death={tunneller.death}
          />
          <ProfileSources sources={tunneller.sources} />
          <ProfileImageSource source={tunneller.image?.source} />
          <HowToCite
            id={tunneller.id}
            summary={tunneller.summary}
            today={today}
          />
        </div>
      </div>
    </>
  );
}
