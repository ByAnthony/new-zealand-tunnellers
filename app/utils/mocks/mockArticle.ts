import { AboutUsArticle } from "../../types/article";

export const mockArticle: AboutUsArticle = {
  id: "about-us",
  title: "Our Story",
  section: [
    { title: "Title 1", text: "This is the first paragraph of our story." },
    { title: "Title 2", text: "This is the second paragraph of our story." },
  ],
  image: [
    {
      file: "our-story.jpg",
      title: null,
      photographer: null,
      reference: null,
      alt: "A picture of our team",
    },
  ],
};
