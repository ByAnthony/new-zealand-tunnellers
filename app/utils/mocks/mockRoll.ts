import { Tunneller } from '@/app/types/roll';
import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails: Tunneller = {
  id: mockId,
  name: johnDoe,
  birthDate: '1886',
  deathDate: '1952',
};

export const mockDetailsList: Tunneller[] = [
  mockDetails,
  {
    ...mockDetails,
    id: 1,
    name: alexanderDriscott,
  },
];

export const mockRoll: Record<string, Tunneller[]> = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
