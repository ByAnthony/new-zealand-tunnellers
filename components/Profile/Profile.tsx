"use client";

import Link from "next/link";

import { HowToCite } from "@/components/HowToCite/HowToCite";
import { ProfileDiary } from "@/components/Profile/ProfileDiary/ProfileDiary";
import { ProfileImageSource } from "@/components/Profile/ProfileImageSource/ProfileImageSource";
import { ProfileSources } from "@/components/Profile/ProfileSources/ProfileSources";
import { ProfileSummary } from "@/components/Profile/ProfileSummary/ProfileSummary";
import { Title } from "@/components/Title/Title";
import { TunnellerProfile } from "@/types/tunneller";
import { displayBiographyDates } from "@/utils/helpers/roll";

import STYLES from "./Profile.module.scss";

type Props = {
  tunneller: TunnellerProfile;
};

export function Profile({ tunneller }: Props) {
  return (
    <>
      <div className={STYLES.container}>
        <div className={STYLES.header}>
          <div className={STYLES.link}>
            <Link href="/tunnellers">Tunnellers</Link>
          </div>
          <Title
            name={tunneller.summary.name}
            subTitle={displayBiographyDates(
              tunneller.summary.birth,
              tunneller.summary.death,
            )}
          />
        </div>
      </div>
      <div className={STYLES.profile}>
        <div className={STYLES["flex-summary"]}>
          <div className={STYLES.summary}>
            <ProfileSummary
              summary={tunneller.summary}
              embarkationUnit={tunneller.militaryYears.embarkationUnit}
              enlistment={tunneller.militaryYears.enlistment}
              image={tunneller.image}
            />
          </div>
        </div>
        <div className={STYLES["flex-diary"]}>
          <ProfileDiary
            tunnellerId={tunneller.id}
            origins={tunneller.origins}
            preWarYears={tunneller.preWarYears}
            militaryYears={tunneller.militaryYears}
            death={tunneller.death}
          />
          <ProfileSources sources={tunneller.sources} />
          <ProfileImageSource source={tunneller.image?.source} />
          <HowToCite id={tunneller.id} summary={tunneller.summary} />
        </div>
      </div>
    </>
  );
}
