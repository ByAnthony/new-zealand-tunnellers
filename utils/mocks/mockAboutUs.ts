import { AboutUsArticle } from "@/types/article";

import {
  mockSectionOne,
  mockSectionTwo,
  mockTitle,
  mockTopImage,
} from "./mockArticle";

export const mockAboutUs: AboutUsArticle = {
  id: "my-awesome-article",
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
