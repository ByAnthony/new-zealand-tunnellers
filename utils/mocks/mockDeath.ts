import { Death } from "@/types/tunneller";

import { mockDeath } from "./mockTunneller";

export const mockAfterWarDeath: Death = {
  ...mockDeath,
  warInjuriesDeathAfterWar: true,
};
