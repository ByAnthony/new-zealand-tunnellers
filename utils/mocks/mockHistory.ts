import { HistoryChapter } from "@/types/homepage";

export const mockHistory: HistoryChapter[] = [
  {
    id: "where-all-started",
    chapter: 1,
    title: "Where All\\Started",
    image: "where-all-started.jpg",
  },
  {
    id: "the-journey-ahead",
    chapter: 2,
    title: "The Journey\\Ahead",
    image: "the-journey-ahead.jpg",
  },
  {
    id: "discovering-the-war",
    chapter: 3,
    title: "Discovering\\The War",
    image: "discovering-the-war.jpg",
  },
  {
    id: "below-the-surface",
    chapter: 4,
    title: "Below\\The Surface",
    image: "below-the-surface.jpg",
  },
  {
    id: "back-to-the-surface",
    chapter: 5,
    title: "Back\\To The Surface",
    image: "back-to-the-surface.jpg",
  },
  {
    id: "reconversion-of-activities",
    chapter: 6,
    title: "Reconversion\\Of Activities",
    image: "reconversion-of-activities.jpg",
  },
  {
    id: "returnin-home",
    chapter: 7,
    title: "Returning\\Home",
    image: "returning-home.jpg",
  },
];
