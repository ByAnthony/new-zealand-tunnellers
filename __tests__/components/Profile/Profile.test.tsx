import { render } from "@testing-library/react";
import { Profile } from "../../../app/components/Profile/Profile";
import { mockTunnellerProfile } from "../../mocks/mockTunneller";

describe("Profile", () => {
  test("matches the snapshot", () => {
    const { asFragment } = render(<Profile tunneller={mockTunnellerProfile} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
