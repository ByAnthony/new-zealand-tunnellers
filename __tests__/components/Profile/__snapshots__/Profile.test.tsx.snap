// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="link"
      >
        <a
          href="/tunnellers"
        >
          Tunnellers
        </a>
      </div>
      <div
        class="main-title"
      >
        <h1>
          <span
            class="forename"
          >
            John 
          </span>
          <span
            class="surname"
          >
            Smith
          </span>
        </h1>
      </div>
      <h2
        class="title-line-3"
      >
        1886-1966
      </h2>
    </div>
  </div>
  <div
    class="profile"
  >
    <div
      class="flex-summary"
    >
      <div
        class="summary"
      >
        <div
          class="overview"
        >
          <div
            class="image-card"
          >
            <img
              alt="Portrait of John Smith"
              class="tunneller-image"
              data-nimg="1"
              decoding="async"
              height="475"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Froll%2Ftunnellers%2F1%2F1234.jpg&w=256&q=75"
              srcset="/_next/image?url=%2Fimages%2Froll%2Ftunnellers%2F1%2F1234.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Froll%2Ftunnellers%2F1%2F1234.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="100"
            />
          </div>
          <div
            class="fullwidth-main-card"
          >
            <p>
              Unit
            </p>
            <span>
              Main Body
            </span>
          </div>
          <div
            class="fullwidth-main-card"
          >
            <p>
              Section
            </p>
            <span>
              Section No.2
            </span>
          </div>
          <div
            class="halfwidth-cards-container"
          >
            <div
              class="halfwidth-secondary-card"
            >
              <p>
                Rank
              </p>
              <span>
                Sapper
              </span>
            </div>
            <div
              class="halfwidth-secondary-card"
            >
              <p>
                Serial
              </p>
              <span>
                1/1000
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex-diary"
    >
      <div
        class="diary"
      >
        <h2>
          About
        </h2>
        <div
          class="fullwidth-main-card"
        >
          <p>
            Born in New Zealand
          </p>
          <span>
            18 December 1886
          </span>
        </div>
        <div
          class="fullwidth-main-card"
        >
          Parents
        </div>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <p>
              Mother
            </p>
            <span>
              Jane Doe
            </span>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <p>
              Father
            </p>
            <span>
              John Doe
            </span>
          </div>
        </div>
        <div
          class="fullwidth-main-card"
        >
          Work
        </div>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Occupation
              </p>
            </div>
            <div>
              <span>
                Goldminer
              </span>
            </div>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <div
              class="halfwidth-secondary-card-title"
            >
              <p>
                Employer
              </p>
            </div>
            <div>
              <span>
                Goldmining Company
              </span>
            </div>
          </div>
        </div>
        <h2>
          Army Experience
        </h2>
        <li
          class="fullwidth-secondary-card"
        >
          <span>
            NZ Infantry
          </span>
          <p
            class="line-margin"
          >
            12 months in New Zealand
          </p>
        </li>
        <a
          aria-label="Open the World War I timeline"
          class="war-service"
          href="/tunnellers/1/wwi-timeline"
        >
          <div>
            <p>
              World War I (1914-1918)
            </p>
            <span>
              New Zealand Tunnellers
            </span>
          </div>
          <div
            class="arrow"
          >
            →
          </div>
        </a>
        <div
          class="halfwidth-cards-container"
        >
          <div
            class="halfwidth-secondary-card"
          >
            <p>
              <img
                alt="British War Medal ribbon"
                data-nimg="1"
                decoding="async"
                height="11"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75"
                srcset="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Froll%2Fmedals%2Fbritish-war-medal.png&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </p>
            <span>
              British War Medal
            </span>
          </div>
          <div
            class="halfwidth-secondary-card"
          >
            <p>
              <img
                alt="Victory Medal ribbon"
                data-nimg="1"
                decoding="async"
                height="11"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75"
                srcset="/_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Froll%2Fmedals%2Fvictory-medal.png&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </p>
            <span>
              Victory Medal
            </span>
          </div>
        </div>
        <h2>
          Death
        </h2>
        <div
          class="fullwidth-main-card"
        >
          <p>
            Died at the age of 89
          </p>
          <span>
            13 October 1926
          </span>
        </div>
      </div>
      <div
        class="sources"
      >
        <h3>
          Sources
        </h3>
        <p>
          Auckland War Memorial Museum Tāmaki Paenga Hira: 
          <a
            href="https://www.mockurl.co.nz/online-cenotaph/B2874930"
          >
            Online Cenotaph He Toa Taumata Rau
          </a>
          .
        </p>
        <p>
          New Zealand Archives Te Rua Mahara o te Kāwanatanga, AABK 18805 W5530 39/0022386, 
          <a
            href="https://www.mockurl.co.nz/nzarchives/B2874930"
          >
            Military Personnel File
          </a>
          .
        </p>
        <p>
          <em>
            London Gazette, 
          </em>
          23 May 1917, p. 1675.
        </p>
        <p>
          <em>
            Nominal Roll of New Zealand Expeditionary Force, 1915. New Zealand Engineers Tunnelling Company
          </em>
          , Government Printer, Wellington, 1916, 37.
        </p>
      </div>
      <div
        class="sources"
      >
        <h3>
          Photograph
        </h3>
        <p>
          Auckland Libraries Ngā Pātaka Kōrero o Tāmaki Makaurau, Sir George Grey Special Collections: 
          <a
            href="https://digitalnz.org/records?text=31-B2671&tab=Images#"
          >
            31-B2671
          </a>
          .
        </p>
      </div>
      <div
        class="howtocite"
      >
        <h3>
          How to cite this page
        </h3>
        <p>
          Anthony Byledbal, “John Smith (1886-1966)“,
          <em>
             New Zealand Tunnellers Website
          </em>
          , 2023 (2009), Accessed: 4 May 2023. 
          <span>
            URL: www.
            <wbr />
            nztunnellers
            <wbr />
            .com/
            <wbr />
            tunnellers/
            <wbr />
            1.
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
